internal/storage/postgres_storage.go

```diff
@@ -0,0 +1,130 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ivanlp-p/ShortLinkService/internal/models"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type PostgresStorage struct {
+	pool *pgxpool.Pool
+}
+
+func NewPostgresStorage(ctx context.Context, dsn string) (Storage, error) {
+	pool, err := pgxpool.New(ctx, dsn)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create connection pool: %w", err)
+	}
+
+	if err := pool.Ping(ctx); err != nil {
+		return nil, fmt.Errorf("unable to ping database: %w", err)
+	}
+
+	if err = createTables(ctx, pool); err != nil {
+		return nil, fmt.Errorf("failed to create tables: %w", err)
+	}
+
+	return &PostgresStorage{pool: pool}, nil
+}
+
+func createTables(ctx context.Context, pool *pgxpool.Pool) error {
+	conn, err := pool.Acquire(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to acquire connection: %w", err)
+	}
+
+	defer conn.Release()
+
+	_, err = conn.Exec(ctx,
+		`CREATE TABLE IF NOT EXISTS urls (uuid TEXT primary key, short_url TEXT NOT NULL UNIQUE,original_url TEXT NOT NULL);`)
+	if err != nil {
+		return fmt.Errorf("unable to acquire connection: %w", err)
+	}
+
+	_, err = conn.Exec(ctx,
+		`CREATE UNIQUE INDEX IF NOT EXISTS idx_original_url_unique ON urls (original_url);`)
+	if err != nil {
+		return fmt.Errorf("failed to create unique index: %w", err)
+	}
+
+	return err
+}
+
+func (p PostgresStorage) LoadFromFile() error {
+	return nil
+}
+
+func (p PostgresStorage) PutOriginalURL(ctx context.Context, shortLink models.ShortLink) error {
```

Comment:
Стоит использовать указатель в качестве ресивера во-избежании нежелательного копирования.
```suggestion
func (p *PostgresStorage) PutOriginalURL(ctx context.Context, shortLink models.ShortLink) error {
```